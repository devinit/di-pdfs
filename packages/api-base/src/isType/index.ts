/**
 * Partially Adopted and modified from https://github.com/justinlettau/ts-util-is
 */
/**
 * Determines if a reference is an `Array`.
 *
 * @param value Reference to check.
 */
export function isArray(value: any): value is any[] {
    return Array.isArray(value);
}

export function isPromise(value: any): boolean {
    return value instanceof Promise;
}

/**
 * Determines if a reference is a `Boolean`.
 *
 * @param value Reference to check.
 */
export function isBoolean(value: any): value is boolean {
    return typeof value === 'boolean';
}

/**
 * Determines if a reference is a `Date`.
 *
 * @param value Reference to check.
 */
export function isDate(value: any): value is Date {
    return Object.prototype.toString.call(value) === '[object Date]';
}

/**
 * Determines if a reference is a valid `Date`.
 *
 * @param value Reference to check.
 */
export function isDateValid(value: any): value is Date {
    return isDate(value) && !isNaN(value.getTime());
}

/**
 * Determines if a reference is defined.
 *
 * @param value Reference to check.
 */
export function isDefined(value: any): boolean {
    return typeof value !== 'undefined';
}

/**
 * Determines if a reference is an `Error`.
 *
 * @param value Reference to check.
 */
export function isError(value: any): value is Error {
    return Object.prototype.toString.call(value) === '[object Error]' || value instanceof Error;
}

/**
 * Determines if a reference is a `Function`.
 *
 * @param value Reference to check.
 */
// tslint:disable-next-line:ban-types
export function isFunction(value: any): value is Function {
    return typeof value === 'function';
}

/**
 * Determines if a reference is `null`.
 *
 * @param value Reference to check.
 */
export function isNull(value: any): value is null {
    return value === null;
}

/**
 * Determines if a reference is a `Number`.
 *
 * @param value Reference to check.
 */
export function isNumber(value: any): value is number {
    return typeof value === 'number';
}

/**
 * Determines if a reference is an 'Object'.
 *
 * @param value Reference to check.
 */
// tslint:disable-next-line:ban-types
export function isObject(value: any): value is Object {
    return typeof value === 'object';
}

/**
 * Determines if a reference is a `String`.
 *
 * @param value Reference to check.
 */
export function isString(value: any): value is string {
    return typeof value === 'string';
}

/**
 * Determines if a reference is `undefined`.
 *
 * @param value Reference to check.
 */
export function isUndefined(value: any): value is undefined {
    return typeof value === 'undefined';
}
