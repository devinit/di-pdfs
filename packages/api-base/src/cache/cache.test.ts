import 'jest';
import {readCacheData, ICached, isKeyInCacheFile} from '.';
import * as prettyFormat from 'pretty-format';
import {isError} from '../isType';
import * as path from 'path';

describe('cache module tests', () => {
    const cachePath = path.resolve(__dirname, './test-data/.cache');

    it('should read precache file and return an array of JS objects', async () => {
        const cachedData: ICached[] | Error = await readCacheData(cachePath);
        if (isError(cachedData)) return console.error(cachedData);
        expect(cachedData[0].key).toBe('global-picture/themes.csv');
        expect(cachedData.length).toBe(3);
        expect(prettyFormat(cachedData)).toMatchSnapshot();
    });

    it ('should write to cache file only unique new keys', async () => {
        const isKeyInCache = await isKeyInCacheFile('global-picture/themes.csv', cachePath);
        if (isError(isKeyInCache)) console.error(isKeyInCache);
        expect(isKeyInCache).toBe(true);
    });
});
