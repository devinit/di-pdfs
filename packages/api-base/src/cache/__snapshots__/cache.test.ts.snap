// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cache module tests should read precache file and return an array of JS objects 1`] = `
"Array [
  Object {
    \\"key\\": \\"global-picture/themes.csv\\",
    \\"type\\": \\"cms\\",
  },
  Object {
    \\"key\\": \\"global/entity.csv\\",
    \\"type\\": \\"cms\\",
  },
  Object {
    \\"key\\": \\"SELECT * FROM data_series.depth_of_extreme_poverty_190 WHERE year >= 2000 AND year <= 2015\\",
    \\"type\\": \\"dw\\",
  },
]"
`;
